battle-service/src/main/java/com/company/battle/controllers/GameController.java

```diff
@@ -0,0 +1,43 @@
+package com.company.battle.controllers;
+
+import com.company.common.dtos.GameDto;
+import com.company.common.dtos.SettingsDto;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+import java.util.UUID;
+
+@RestController
+@RequestMapping("api/v1/game")
+public class GameController {
+
+    @GetMapping("/{id}")
+    GameDto getGameById(@PathVariable UUID id) {
+        return null;
+    }
+
+    @GetMapping("/{link}")
+    GameDto getGameByLink(@PathVariable String link) {
+        return null;
+    }
+
+    @GetMapping
+    List<GameDto> getActiveGames() {
+        return null;
+    }
+
+    @PostMapping("/create")
+    GameDto createGame(@RequestBody GameDto gameDto) {
+        return null;
+    }
+
+    @PutMapping("/join")
+    GameDto joinGame(@RequestBody String link) {
```

Comment:
@RequestBody String link is not by REST


you can make @PutMapping("/join/{link}")